substrate/frame/opf/src/lib.rs

```diff
@@ -0,0 +1,363 @@
+		MaximumProjectsNumber,
+
+		/// This voting round does not exists
+		NoRoundFound,
+
+		/// Voting period closed for this round
+		VotePeriodClosed,
+
+		/// Not enough funds to vote, you need to decrease your stake
+		NotEnoughFunds,
+
+		/// The reward calculation failed due to an internal error
+		FailedRewardCalculation,
+
+		/// Voting round is over
+		VotingRoundOver,
+
+		/// User's funds still cannot be unlocked
+		FundsUnlockNotPermitted,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Weight: see `begin_block`
+		fn on_idle(n: BlockNumberFor<T>, remaining_weight: Weight) -> Weight {
+			Self::on_idle_function(n, remaining_weight)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// OPF voting logic
+		///
+		/// ## Dispatch Origin
+		///
+		/// Must be signed
+		///
+		/// ## Details
+		///
+		/// This extrinsic allows users to [vote for/nominate] a whitelisted project using their
+		/// funds. As a first implementation, the `conviction` parameter was not included for
+		/// simplicity, but /// should be in the next iteration of the pallet. The amount defined
+		/// by the user is locked and released only when the project reward is /// sent for
+		/// distribution, or when the project is not dimmed fundable. Users can edit an existing
+		/// vote within the vote-casting period. Then, during the vote-locked period, rewards are
+		/// calculated based on the total user amount attributed to each project by the user’s
+		/// votes.
+		///
+		/// ### Parameters
+		/// - `project_id`: The account that will receive the reward.
+		/// - `amount`: Amount that will be locked in user’s balance to nominate a project.
+		/// - `is_fund`: Parameter that defines if user’s vote is in favor (*true*), or against
+		///   (*false*)
+		/// the project funding.
+
+		/// ### Errors
+		/// - [`Error::<T>::NotEnoughFunds`]: The user does not have enough balance to cast a vote
+		///  
+		/// ## Events
+		#[pallet::call_index(0)]
+		#[pallet::weight(<T as Config>::WeightInfo::vote(T::MaxWhitelistedProjects::get()))]
+		pub fn vote(
+			origin: OriginFor<T>,
+			project_id: ProjectId<T>,
+			#[pallet::compact] amount: BalanceOf<T>,
+			is_fund: bool,
+			conviction: Conviction,
+		) -> DispatchResult {
+			let voter = ensure_signed(origin)?;
+			// Get current voting round & check if we are in voting period or not
+			Self::period_check()?;
+			// Check that voter has enough funds to vote
+			let voter_balance = T::NativeBalance::total_balance(&voter);
+			ensure!(voter_balance > amount, Error::<T>::NotEnoughFunds);
+
+			// Check the total amount locked in other projects
+			let voter_holds = BalanceOf::<T>::zero();
+			let projects = WhiteListedProjectAccounts::<T>::get();
+			for project in projects {
+				let infos = Votes::<T>::get(&project, &voter);
+				let this_amount = infos.unwrap_or_default().amount;
+				voter_holds.saturating_add(this_amount);
```

Comment:
```suggestion
				if let Some(infos) = Votes::<T>::get(&project, &voter) {
    				voter_holds.saturating_add(infos.amount);				
				}
```

or `let this_amount = infos.map(|x| x.amount).unwrap_or_default()`

so that no need to provide dummy Default implementation